sentry/src/main/java/io/sentry/metrics/MetricsHelper.java

```diff
@@ -0,0 +1,222 @@
+    final @NotNull String typePrefix = toStatsdType(type);
+    final @NotNull String serializedTags = GetTagsKey(tags);
+
+    final @NotNull String unitName = getUnitName(unit);
+    return String.format("%s_%s_%s_%s", typePrefix, metricKey, unitName, serializedTags);
+  }
+
+  @NotNull
+  private static String getUnitName(final @Nullable MeasurementUnit unit) {
+    return (unit != null) ? unit.apiName() : MeasurementUnit.NONE;
+  }
+
+  private static String GetTagsKey(final @Nullable Map<String, String> tags) {
+    if (tags == null || tags.isEmpty()) {
+      return "";
+    }
+
+    final @NotNull StringBuilder builder = new StringBuilder();
+    for (Map.Entry<String, String> tag : tags.entrySet()) {
+
+      // Escape delimiters in key and value
+      final @NotNull String key = escapeString(tag.getKey());
+      final @NotNull String value = escapeString(tag.getValue());
+
+      if (builder.length() > 0) {
+        builder.append(TAGS_PAIR_DELIMITER);
+      }
+
+      builder.append(key).append(TAGS_KEY_VALUE_DELIMITER).append(value);
+    }
+
+    return builder.toString();
+  }
+
+  @NotNull
+  private static String escapeString(final @NotNull String input) {
+    final StringBuilder escapedString = new StringBuilder(input.length());
+
+    for (int idx = 0; idx < input.length(); idx++) {
+      final char ch = input.charAt(idx);
+
+      if (ch == TAGS_PAIR_DELIMITER || ch == TAGS_KEY_VALUE_DELIMITER) {
+        escapedString.append(TAGS_ESCAPE_CHAR); // Prefix with escape character
+      }
+      escapedString.append(ch);
+    }
+
+    return escapedString.toString();
+  }
+
+  public static double convertNanosTo(
+      final @NotNull MeasurementUnit.Duration unit, final long durationNanos) {
+    switch (unit) {
+      case NANOSECOND:
+        return durationNanos;
+      case MICROSECOND:
+        return (double) durationNanos / 1000.0d;
+      case MILLISECOND:
+        return (double) durationNanos / 1000000.0d;
+      case SECOND:
+        return (double) durationNanos / 1000000000.0d;
+      case MINUTE:
+        return (double) durationNanos / 60000000000.0d;
+      case HOUR:
+        return (double) durationNanos / 3600000000000.0d;
+      case DAY:
+        return (double) durationNanos / 86400000000000.0d;
+      case WEEK:
+        return (double) durationNanos / 86400000000000.0d / 7.0d;
+      default:
+        throw new IllegalArgumentException("Unknown Duration unit: " + unit.name());
+    }
+  }
+
+  /**
+   * Encodes the metrics
+   *
+   * @param timestamp The bucket time the metrics belong to, in second resolution
+   * @param metrics The metrics to encode
+   * @param writer The writer to encode the metrics into
+   */
+  public static void encodeMetrics(
+      final long timestamp,
+      final @NotNull Collection<Metric> metrics,
+      final @NotNull StringBuilder writer) {
+    for (Metric metric : metrics) {
+      writer.append(sanitizeKey(metric.getKey()));
```

Comment:
can you add a comment with the reference of the sintax used?

Reply:
add a reference to https://getsentry.github.io/relay/relay_metrics/index.html, too, please